src/hw/driver/virt_serial.c

```diff
@@ -0,0 +1,22 @@
+#include "common/serial_port.h"
+#include <stdint.h>
+
+SerialPrint platform_serial;
+
+static volatile uint32_t* const UART0 = (uint32_t*)UART_BASE;
```

Comment:
We could remove the cmake defined UART_BASE here by doing something like:
```
#ifdef __aarch64__
#define UART_BASE <the aarch64 value>;
#elif defined __thumb__
#define UART_BASE <the thumb value>
#elif defined __arm__
#define UART_BASE <the arm value>
#else
#error No virt serial port base address for this architecture!
#endif

static volatile uint32_t* const UART0 = (uint32_t*)UART_BASE;
```
Since that base is only relevant to virt.